modules/local/novaseq_err.nf

```diff
@@ -13,16 +13,29 @@ process NOVASEQ_ERR {
     tuple val(meta), path("*.md.err.rds"), emit: errormodel
     tuple val(meta), path("*.md.err.pdf"), emit: pdf
     tuple val(meta), path("*.md.err.convergence.txt"), emit: convergence
-    //path "versions.yml"                  , emit: versions
+    path "versions.yml"                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
 
     script:
     if (!meta.single_end) {
         """
         novaseq_err_pe.r ${errormodel[0]} ${errormodel[1]} ${meta.run}
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            R: \$(R --version | sed -n 1p | sed 's/R version //g' | sed 's/\\s.*\$//')
+        END_VERSIONS
         """
     } else {
         """
         novaseq_err_se.r ${errormodel} ${meta.run}
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            R: \$(R --version | sed -n 1p | sed 's/R version //g' | sed 's/\\s.*\$//')
+        END_VERSIONS
```

Comment:
The two version calls are identical, right? If so, one block after the if clause would do it, I guess.

Reply:
Yes, the version calls are identical. However, they are in groovy if/else and separate bash blocks, so I am actually not sure whether that works after the if clause. I'll test on Friday!

Reply:
I see. Not worth it I suppose.